waiter/test/waiter/service_description_test.clj

```diff
@@ -2199,6 +2199,25 @@
         constraints-schema profile->defaults config
         "The health check port index (5) must be smaller than ports (3)"))
 
+    (testing "testing invalid backend proto and health check proto combination"
+      (let [supported-protocols #{"http" "https" "h2c" "h2"}]
+        (doseq [backend-proto supported-protocols]
+          (doseq [health-check-proto (disj supported-protocols backend-proto)]
```

Comment:
These two `doseq`s can also be combined.

Reply:
Done.